src/drivers/cmfileapi-driver.ts

```diff
@@ -0,0 +1,218 @@
+      platform: cache.get('CMAKE_GENERATOR_TOOLSET') ? cache.get('CMAKE_GENERATOR_TOOLSET')!.value : undefined
+    } as CMakeGenerator;
+
+    this._generatorInformation = {
+      name: cache.get('CMAKE_GENERATOR')!.value,
+      platform: cache.get('CMAKE_GENERATOR_TOOLSET') ? cache.get('CMAKE_GENERATOR_TOOLSET')!.value : undefined
+    };
+  }
+
+  async doInit() {
+    // The seams to be a different between server mode and fileapi on load of a existing project
+    // If the existing project is not generated by the IDE then the fileapi queries are missing.
+    // but the generator information are needed to get the code mode, cache and cmake files.
+    // This workaround load the information from cache.
+    if (await fs.exists(this.cachePath)) {
+      await this.loadGeneratorInformationFromCache(this.cachePath);
+      await this.doConfigure([], undefined);
+    } else {
+      this._generatorInformation = this.generator;
+    }
+    if (!this.generator) {
+      throw new NoGeneratorError();
+    }
+
+    this._cacheWatcher.onDidChange(() => {
+      log.debug(`Reload CMake cache: ${this.cachePath} changed`);
+      rollbar.invokeAsync('Reloading CMake Cache', () => this._reloadPostConfigure());
+    });
+  }
+
+  doConfigureSettingsChange() { this._needsReconfigure = true; }
+  async checkNeedsReconfigure(): Promise<boolean> { return this._needsReconfigure; }
+
+  async doSetKit(need_clean: boolean, cb: () => Promise<void>): Promise<void> {
+    this._needsReconfigure = true;
+    if (need_clean) {
+      await this._cleanPriorConfiguration();
+    }
+    await cb();
+    if (!this.generator) {
+      throw new NoGeneratorError();
+    }
+  }
+
+  async asyncDispose() {
+    this._codeModelChanged.dispose();
+    this._cacheWatcher.dispose();
+  }
+
+  protected async doPreCleanConfigure(): Promise<void> { await this._cleanPriorConfiguration(); }
+
+  async doConfigure(args_: string[], outputConsumer?: proc.OutputConsumer): Promise<number> {
+    const api_path = this.getCMakeFileApiPath();
+    await createQueryFileForApi(api_path);
+
+    // Dup args so we can modify them
+    const args = Array.from(args_);
+    args.push('-H' + util.lightNormalizePath(this.sourceDir));
+    const bindir = util.lightNormalizePath(this.binaryDir);
+    args.push('-B' + bindir);
+    const gen = this.generator;
+    if (gen) {
+      args.push(`-G${gen.name}`);
+      if (gen.toolset) {
+        args.push(`-T${gen.toolset}`);
+      }
+      if (gen.platform) {
+        args.push(`-A${gen.platform}`);
+      }
+    }
+    const cmake = this.cmake.path;
+    log.debug('Invoking CMake', cmake, 'with arguments', JSON.stringify(args));
+    const env = await this.getConfigureEnvironment();
+    const res = await this.executeCommand(cmake, args, outputConsumer, {environment: env}).result;
+    log.trace(res.stderr);
+    log.trace(res.stdout);
+    if (res.retc == 0) {
+      this._needsReconfigure = false;
+    }
+    await this._reloadPostConfigure();
```

Comment:
Does it make sense to call `_reloadPostConfigure` if the configure failed?  Should this be in the `if (res.retc == 0)` block?

Reply:
I changed it. CMake only generates a code model if it runs successfully.
Additionally, I changed the behavior of the `doInit`-function. The `doInit`-function runs on driver creation. It is not necessary to configure if there is already a CodeModel is present. But this brings the driver into an inconsistent state between extension expected kit and the used kit/settings for the generation of CMake cache.